content/client-lib-development-guide/features.textile

```diff
@@ -870,6 +871,7 @@ h4. Message
 ** @(TM2g)@ @name@ string
 ** @(TM2d)@ @data@ string, buffer or JSON-encodable object or array
 ** @(TM2e)@ @encoding@ string
+** @(TM2i)@ @extras@ JSON-like object, used to contain any arbitrary key value pairs which may also contain other primitive JSON types, JSON-like objects or JSON-like arrays. For context, the @extras@ field is planned for us with push notifications as a payload in the @push@ key, and later to allow arbitrary headers in the @header@ key that may be used for routing, filtering or to expose to consumers when retrieving messages from queues
```

Comment:
AFAIK we haven't decided to include a `headers` member so we should not say this. Just say that other message metadata or ancillary payloads may be included in `extras` in the future.

Also, call me pedantic but I don't like the expression "JSON-like object" because objects are objects and JSON is text. We could say "JSON-encodable object" or simply "map".